kernel/src/log_segment.rs

```diff
@@ -478,28 +483,99 @@ impl LogSegment {
         )?))
     }
 
-    // Do a lightweight protocol+metadata log replay to find the latest Protocol and Metadata in
-    // the LogSegment
+    // Find the latest Protocol and Metadata in the LogSegment. This is done by replaying the log
+    // until the latest_crc_file/latest checkpoint is reached, at which point we can just return the
+    // P+M from the CRC directly or read from checkpoint.
+    //
+    // Note that according to the spec CRC files must contain the latest Protocol and Metadata. If
```

Comment:
cc @huan233usc

Reply:
technically speaking ---- very old table(table written before) https://github.com/delta-io/delta/commit/52716a44ba236ab3973f9ebfc0fcb0532b08c3a2 might without P&M

worth consider if you want to take this case into account.